src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreOptionFactory.cs

```diff
@@ -66,6 +74,11 @@ public void UpdateOption(string optionKey, IRestoreDatabaseTaskDataObject restor
                 var defaultValue = builder.DefaultValueFunction(restoreDataObject);
                 var validateResult = builder.ValidateFunction(restoreDataObject, currentValue, defaultValue);
                 optionInfo.IsReadOnly = validateResult.IsReadOnly;
+                if (optionInfo.IsReadOnly)
+                {
+                    //TODO: should we set the current value to default if the option is read only?
+                    //optionInfo.CurrentValue = defaultValue;
+                }
```

Comment:
should we check in the code like this?  what are we wanting for to answer the question in the comment?